src/Recruiter/Job/Repository.php

```diff
@@ -241,6 +241,181 @@ public function recentHistory($group = null, T\Moment $at = null, array $query =
         ];
     }
 
+    /**
+     * @param int
+     */
+    public function countSlowRecentJobs(
+        T\Moment $lowerLimit,
+        T\Moment $upperLimit,
+        $secondsToConsiderJobAsSlow=5
+    )
+    {
+        $archived = $this->archived->aggregate([
+            [
+                '$match' => [
+                    'last_execution.ended_at' => [
+                        '$gte' => T\MongoDate::from($lowerLimit),
+                    ],
+                ]
+            ],
+            [
+                '$project' => [
+                    '_id' => '$_id',
+                    'execution_time' => [
+                        '$subtract' => [
+                            '$last_execution.ended_at',
+                            '$last_execution.started_at'
+                        ]
+                    ],
+                ]
+            ],
+            [
+                '$match' => [
+                    'execution_time' => [
+                        '$gt' => $secondsToConsiderJobAsSlow*1000 
+                    ]
+                ],
+            ],
+        ])['result'];
+
+        return count($archived) + $this->scheduled->count([
+            'scheduled_at' => [
+                '$gte' => T\MongoDate::from($lowerLimit),
+                '$lte' => T\MongoDate::from($upperLimit)
+            ],
+            'last_execution.started_at' => [
+                '$exists' => true,
+            ],
+            'last_execution.ended_at' => [
+                '$exists' => true,
+            ],
+        ]);
+    }
+
+    /**
+     * @param int
+     */
+    public function countRecentJobsWithManyAttempts(T\Moment $lowerLimit)
+    {
+        $archived = $this->archived->count([
+            'last_execution.ended_at' => [
+                '$gte' => T\MongoDate::from($lowerLimit),
+            ],
+            'attempts' => [
+                '$gt' => 1
+            ]
+        ]);
+        $scheduled = $this->scheduled->count([
+            'attempts' => [
```

Comment:
I think it's better to add a filter on the last_execution also in scheduled jobs

Reply:
Since we are interested to the recency of the data, I think it's better to always have a `lowerLimit` and an `upperLimit` to apply to both `archived` and `scheduled`